extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -284,28 +285,25 @@ public Response doGetResults(
       }
       final AuthenticationResult authenticationResult = AuthorizationUtils.authenticationResultFromRequest(req);
 
-      if (offset != null && offset < 0) {
+      if (page == null) {
```

Comment:
Page==null means we need to fetch all the pages. So this validation needs to be removed.